### PR TITLE
{bio,devel,vis}[GCCcore/12.3.0] QuPath v0.5.0, Gradle v8.6, OpenSlide-Java v0.12.4 w/ Java 17

### PR DESCRIPTION
**Pull request by PetrKralCZ**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/g/Gradle/Gradle-8.6.eb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+"""
+
+toolchain = SYSTEM
+
+source_urls = ['https://services.gradle.org/distributions']
+sources = ['gradle-%(version)s-all.zip']
+checksums = ['85719317abd2112f021d4f41f09ec370534ba288432065f4b477b6a3b652910d']
+
+sanity_check_paths = {
+    'files': ['bin/gradle'],
```
Comment (boegel) [inline easybuild/easyconfigs/g/Gradle/Gradle-8.6.eb:19]: Please include a sanity check command as well:

```
sanity_check_commands = ["gradle --help"]
```

Also, `Gradle` is a Java app, the `gradle` script uses the `java` command, so `Java` should be included as a runtime dependency (and the `versionsuffix` should be set accordingly)

File: easybuild/easyconfigs/o/OpenSlide-Java/OpenSlide-Java-0.12.4-GCCcore-12.3.0-Java-17.eb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+dependencies = [
+    ('OpenSlide', '3.4.1', '-largefiles'),
+    ('Java', '17', '', SYSTEM),
+]
+
+sanity_check_paths = {
+    'files': [
+        'lib/openslide-java/openslide.jar',
+        'lib/openslide-java/libopenslide-jni.so',
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenSlide-Java/OpenSlide-Java-0.12.4-GCCcore-12.3.0-Java-17.eb:29]:
```suggestion
        'lib/openslide-java/libopenslide-jni.%s' % SHLIB_EXT,

File: easybuild/easyconfigs/o/OpenSlide-Java/OpenSlide-Java-0.12.4-GCCcore-12.3.0-Java-17.eb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    ],
+    'dirs': []
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenSlide-Java/OpenSlide-Java-0.12.4-GCCcore-12.3.0-Java-17.eb:31]:
```suggestion
    'dirs': [],

File: easybuild/easyconfigs/q/QuPath/QuPath-0.5.0-GCCcore-12.3.0-Java-17.eb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+files_to_copy = [
+    'build/dist/%(name)s/bin',
+    'build/dist/%(name)s/lib',
+]
+
+modextrapaths = {'PATH': ['']}
+
+sanity_check_paths = {
+    'files': ['bin/%(name)s'],
+    'dirs': ['lib'],
```
Comment (boegel) [inline easybuild/easyconfigs/q/QuPath/QuPath-0.5.0-GCCcore-12.3.0-Java-17.eb:40]: I would make this a bit more specific, by checking for `lib/app` + `lib/runtime`, and also checking for `'lib/libapplauncher.%s' % SHLIB_EXT` in `files` list

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/q/QuPath/QuPath-0.5.0-GCCcore-12.3.0-Java-17.eb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+sanity_check_paths = {
+    'files': [
+        'bin/%(name)s',
+        'lib/libapplauncher.%s' % SHLIB_EXT,
+    ],
+    'dirs': [
+        'lib/app',
+        'lib/runtime',
+    ],
+}
```
Comment (boegel) [inline easybuild/easyconfigs/q/QuPath/QuPath-0.5.0-GCCcore-12.3.0-Java-17.eb:47]:
```suggestion
sanity_check_paths = {
    'files': ['bin/qupath', 'lib/libapplauncher.%s' % SHLIB_EXT],
    'dirs': ['lib/app', 'lib/runtime'],
}
```

File: easybuild/easyconfigs/q/QuPath/QuPath-0.5.0-GCCcore-12.3.0-Java-17.eb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+sanity_check_commands = ['%(name)s --help']
```
Comment (boegel) [inline easybuild/easyconfigs/q/QuPath/QuPath-0.5.0-GCCcore-12.3.0-Java-17.eb:49]:
```suggestion
sanity_check_commands = ["qupath --help"]
```

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node3129.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/ff1715590058c810c7e2e28bf836fd43 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @PetrKralCZ!

**Pull request merged**